powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/converter/RecoveryPukConverter.java

```diff
@@ -161,78 +74,38 @@ public String fromDBValue(RecoveryPuk pukHash, long applicationRid, String userI
      * @return Server private key as composite database value.
      * @throws GenericServiceException Thrown when server private key encryption fails.
      */
-    public RecoveryPuk toDBValue(String pukHash, long applicationRid, String userId, String recoveryCode, long pukIndex) throws GenericServiceException {
-        final String masterDbEncryptionKeyBase64 = powerAuthServiceConfiguration.getMasterDbEncryptionKey();
+    public RecoveryPuk toDBValue(final String pukHash, final long applicationRid, final String userId, final String recoveryCode, final long pukIndex) throws GenericServiceException {
+        final byte[] data = convert(pukHash);
+        final Encryptable encryptable = encryptionService.encrypt(data, createSecretKeyDerivationInput(applicationRid, userId, recoveryCode, pukIndex));
+        return new RecoveryPuk(encryptable.getEncryptionMode(), convert(encryptable));
+    }
 
-        // In case master DB encryption key does not exist, do not encrypt the PUK hash
-        if (masterDbEncryptionKeyBase64 == null || masterDbEncryptionKeyBase64.isEmpty()) {
-            return new RecoveryPuk(EncryptionMode.NO_ENCRYPTION, pukHash);
+    private static byte[] convert(final RecoveryPuk source) {
```

Comment:
😞 Quite complicated but working and backward compatible as proven by the test.
- `fromDBValue` maps a plain string into a plain string (_identity_) in case of NO_ECRYPTION. Otherwise, it decodes base64, decrypts, and creates a plain string.
- `toDBValue` maps a plain string into a plain string (_identity_) in case of NO_ECRYPTION. Otherwise gets bytes from the string, encrypts, and encodes to base64.